Restructuring of directories and information in the wazuh agent and server installation

jmv74211 opened the pull request:
Hello team,

The changes proposed in this PR are as follows:

- [x] Restructured directories of the installation of the Wazuh server.
- [x] Updated documentation of the installation of the agent through sources in Solaris
- [x] Restructured directories of the installation of Wazuh agent. It is now classified by operating systems, and within each one, installation is distinguished by packages and sources.
- [x] Renamed references to packages per package in agent installation.
- [x] Restructured deployment variables section in agent installation. There is now a separate section for each operating system.
- [x] Added agent installation documentation via sources in macOS
- [x] Renamed agent and server installation headers to follow the same nomenclature.
- [x] Renamed agent and server installation files to follow the same nomenclature.
- [x] Replaced the redirections `:doc:` with `:ref:` in the installation files
- [x] Modified and updated all the old references to the new changes produced.

The new documents added about the installation of the agent in Solaris and macOS have been successfully tested in:

- [x] Solaris 11 SPARC
- [x] macOS 1012

alberpilot commented on source/installation-guide/installing-wazuh-agent/solaris/solaris10/wazuh_agent_sources_solaris10.rst:
```diff
@@ -25,7 +25,7 @@ Installing Wazuh agent
          .. code-block:: console
 
-            # pkgutil -y -i git automake gmake autoconf libtool wget curl gcc5core
+            # /opt/csw/bin/pkgutil -y -i git automake gmake autoconf libtool wget curl gcc5core
```
Question: Is the path necessary?

jmv74211 replied:
It depends. It can be used `export PATH=$PATH:/opt/csw/bin:/usr/sfw/bin` and we could eliminate the entire path.

jmv74211 replied:
Done :+1:

alberpilot commented on source/installation-guide/installing-wazuh-agent/solaris/solaris11/index.rst:
```diff
@@ -0,0 +1,24 @@
++---------------------------------------------------------------------------+-----------------------------------------------------------------------+
+| Type                                                                      | Description                                                           |
++===========================================================================+=======================================================================+
+| :ref:`Package installation <wazuh_agent_package_solaris11>`               | Install Wazuh agent Solaris 11 via package.                           |
```
Lets unify "from package" or "via package" in all documents

jmv74211 replied:
Done :+1:

alberpilot commented on source/installation-guide/installing-wazuh-agent/solaris/solaris11/wazuh_agent_package_solaris11.rst:
```diff
@@ -1,36 +1,24 @@
-Install Wazuh agent on Solaris
-===============================
+Installation from package
+==========================
```
Longer than text

jmv74211 replied:
Done :+1:

jmv74211 commented on source/installation-guide/installing-wazuh-agent/aix/wazuh_agent_package_aix.rst:
```diff
@@ -1,19 +1,19 @@
 
-Install Wazuh agent on AIX
-==========================
+Installation from package
```
Done :+1:

jmv74211 commented on source/installation-guide/installing-wazuh-agent/linux/rhel5/wazuh_agent_package_rhel5.rst:
```diff
@@ -1,11 +1,11 @@
+.. _wazuh_agent_package_rhel5:
 
-CentOS/RHEL/Oracle Linux 5
+Red Hat Enterprise Linux 5
```
Done :+1:

jmv74211 commented on source/installation-guide/installing-wazuh-agent/aix/wazuh_agent_package_aix.rst:
```diff
@@ -1,19 +1,19 @@
       # rpm -ivh wazuh-agent-3.9.5-1.aix.ppc.rpm
 
-    With this simple installation, the next step is to register and configure it to communicate with the manager. For more information about this process, please visit the document: :doc:`user manual<../../user-manual/registering/index>`.
+    With this simple installation, the next step is to register and configure it to communicate with the manager. For more information about this process, please visit the document: :doc:`user manual<../../../user-manual/registering/index>`.
```
Done :+1:

jmv74211 commented on source/installation-guide/installing-wazuh-agent/aix/wazuh_agent_package_aix.rst:
```diff
@@ -1,19 +1,19 @@
+=========================
 
-The Wazuh agent for AIX can be downloaded from our :doc:`packages list<../packages-list/index>`. You can choose installation or a deployment:
+The Wazuh agent for AIX can be downloaded from our :doc:`packages list <../../packages-list/index>`. You can choose installation or a deployment:
```
Done :+1:

jmv74211 commented:
Added the following changes:

- [x] Added agent installation documentation via sources in macOS
- [x] Renamed titles of Wazuh server installations

jmv74211 commented:
Added the following changes:

- [x]  Divided the information of each operating system independently in the Wazuh server installation documentation.
- [x]  Updated Filebeat installation table to add each operating system.
- [x]  Renamed names and labels of the installation files by packages of the Wazuh server. Changed rpm --> packages and updated all their references.
